flow_control/static/js/injection.js

```diff
@@ -133,15 +143,25 @@ function FlowControlGoto(runtime, element, options) {
       case actions["show_message"]:
         viewblocks["seqContent"].html(settings["message"]);
         break;
-    }  
+    }
   }
 
   if (runtime.data("runtime-class") === "PreviewRuntime") {
-    
+
+    $("header.xblock-header-check-point li.action-item.action-visibility").hide();
+    $("header.xblock-header-check-point li.action-item.action-duplicate").hide();
+
+  }
+}
+
+function EditFlowControl(runtime, element) {
+
+    // Call to StudioEditableXblockMixin function in order to initialize xblock correctly
+    window.StudioEditableXBlockMixin(runtime, element);
+
+    viewblocks.hideNotAllowedOption();
+
     $("body").on("change","#xb-field-edit-action",function(){
```

Comment:
Are we sure this is not binding every time, so that after a few edits->...->close we are calling this function many simultaneous times?